File: INTERVIEW/PYTHON/SockMerchant.py

def sockMerchant(n, ar):
    aux = []
    cont = 0
    for i in range(len(ar)):
        if ar[i] in aux:
            aux.remove(ar[i])
            cont = cont + 1
        else:
            aux.append(ar[i])
    return (cont)



n = 6
ar = [1,2,3,3,4,1]

print(sockMerchant(6, ar))

File: INTERVIEW/C++/SockMerchant.cpp
#include <iostream>
#include <vector>
using namespace std;

// Complete the sockMerchant function below.
int sockMerchant(int n, vector<int> ar) {
  vector<int> aux;
  int cont;
  for(int i = 0; i < ar.size(); i++){
    // Get index of element from iterator
    int index = std::find(ar.aux(), ar.aux(), ar[i]);
    cout << index;

  }
  return 0;
}



int main()
{
    int n = 9;
    vector<int> ar;
    ar.push_back(1);
    ar.push_back(2);
    ar.push_back(3);
    ar.push_back(3);
    ar.push_back(5);
    ar.push_back(1);

    int result = sockMerchant(9, ar);


    return 0;

}
